src/index.ts

```diff
@@ -63,21 +63,42 @@ interface DecryptJWEParams {
 }
 
 interface AuthParams {
+  nonce: string
+  aud: string
   paths: Array<string>
 }
 
+const sign = async (
+  payload: Record<string, any>,
+  did: string,
+  secretKey: Uint8Array,
+  protectedHeader?: Record<string, any>
+) => {
+  const kid = `${did}#${did.split(':')[2]}`
+  const signer = NaclSigner(u8a.toString(secretKey, B64))
+  const header = toStableObject(Object.assign(protectedHeader || {}, { kid, alg: 'EdDSA' }))
+  return createJWS(toStableObject(payload), signer, header)
+}
+
 const didMethods: HandlerMethods<Context> = {
-  did_authenticate: ({ did }, params: AuthParams) => {
-    return { did, paths: params.paths }
+  did_authenticate: async ({ did, secretKey }, params: AuthParams) => {
+    const response = await sign(
+      {
+        did,
+        aud: params.aud,
+        nonce: params.nonce,
+        paths: params.paths,
+        exp: Math.floor(Date.now() / 1000) + 600, // expires 10 min from now
```

Comment:
Maybe this could be optionally provided in `AuthParams`, in case the consumer may want to change this expiration time?

Reply:
I don't think that should be up to the consumer.

Reply:
There might be different security requirements, no? An OTP system might want to limit the validity to 30 seconds, while an email verification system could leave up to 30 minutes to authenticate for example.

Reply:
Right, I guess this should be addressed in the EIP2844 spec. Let's update it if it becomes a problem.